src/ogkush.js

```diff
@@ -11870,6 +11870,82 @@ class OGInfinity {
     }
   }
 
+  resourceTransportTooltip() {
+    const flyingDetails = {};
+    this.json.flying.ids.forEach((mov) => {
+      if (mov.resDest) {
+        const coords = mov.back ? mov.origin : mov.dest;
+        flyingDetails[coords] = flyingDetails[coords] || {
+          metal: 0,
+          crystal: 0,
+          deuterium: 0,
+        };
+        flyingDetails[coords].metal += mov.metal || 0;
+        flyingDetails[coords].crystal += mov.crystal || 0;
+        flyingDetails[coords].deuterium += mov.deuterium || 0;
+        flyingDetails[coords].name = mov.back ? mov.originName : mov.destName;
+        flyingDetails[coords].own = false;
+      }
+    });
+    this.json.empire.forEach((planet) => {
+      const indexPlanet = planet.coordinates.slice(1, -1) + "P";
+      if (flyingDetails[indexPlanet]) {
+        flyingDetails[indexPlanet].own = true;
+      }
+      if (planet.moon) {
+        const indexMoon = planet.coordinates.slice(1, -1) + "M";
+        if (flyingDetails[indexMoon]) {
+          flyingDetails[indexMoon].own = true;
+        }
+      }
+    });
+
+    const tooltipDiv = DOM.createDOM("div", {}, this.getTranslatedText(128));
+    tooltipDiv.appendChild(DOM.createDOM("div", { class: "splitLine" }));
+    const tableDiv = tooltipDiv.appendChild(DOM.createDOM("table", { class: "flyingFleet" }));
+    const rowHeader = tableDiv.appendChild(DOM.createDOM("tr"));
+    rowHeader.append(
+      DOM.createDOM("th", { colspan: 3 }, this.getTranslatedText(127)),
+      DOM.createDOM("th", { class: "ogl-metal" }, this.getTranslatedText(0, "res")),
+      DOM.createDOM("th", { class: "ogl-crystal" }, this.getTranslatedText(1, "res")),
+      DOM.createDOM("th", { class: "ogl-deut" }, this.getTranslatedText(2, "res"))
+    );
+
+    for (const [coords, details] of Object.entries(flyingDetails)) {
```

Comment:
Because first, all the total resource per planet are computed. The second loop is to build the UI.

![imagen](https://github.com/ogame-infinity/web-extension/assets/13818551/4fe92cf5-dbee-4c34-813f-8b6a2e6d7a89)
![imagen](https://github.com/ogame-infinity/web-extension/assets/13818551/e8854592-2540-4ddb-b302-bc2f557a027b)

May we build all at once just using one loop, but i think maybe gets some cumbersome, and the possible speed gain seems negligible.